tools/depends/download-files.include

```diff
@@ -34,6 +40,18 @@ ifeq ($(HASH_FOUND),no)
 endif
 ifeq ($(HASH_FOUND),yes)
 #	we really need 2 spaces between sha hash and file name!
+#	if the hash sum doesn't match we retry up to 3 times
 	@cd $(TARBALLS_LOCATION); echo "$(HASH_SUM)  $(ARCHIVE)" > $(ARCHIVE).$(HASH_TYPE) && $(HASH_TOOL) $(HASH_TOOL_FLAGS) $(ARCHIVE).$(HASH_TYPE) \
-          || { echo "Error: failed to verify hash sum of $(ARCHIVE), expected type: $(HASH_TYPE) value $(HASH_SUM)"; exit 3; }
+          || {\
+            echo "Error: failed to verify hash sum of $(ARCHIVE), expected type: $(HASH_TYPE) value $(HASH_SUM), retrying.." ;\
+            tries=1 ;\
+            while [ $$tries -le 3 ]; do \
+              echo "download $(ARCHIVE) retry $$tries" ;\
+              rm $(TARBALLS_LOCATION)/$(ARCHIVE) ;\
```

Comment:
Should we be rm'ing $(ARCHIVE).$(HASH_TYPE) as well just for cleanup?

Reply:
it would just be recreated with the exact same content

Reply:
isnt it generated of the downloaded tarball (potentially a bad hash)?
More a safety thing, but all good. happy if things work

Reply:
no, the SHA[512|256] defined in the Makefile or version file is used to create it, not the actual hash_sum of the tarball. That would defy the intention of actually verifying the downloaded tarball ;)